pkg/library/home/persistentHome.go

```diff
@@ -95,3 +101,64 @@ func storageStrategySupportsPersistentHome(workspace *common.DevWorkspaceWithCon
 	storageClass := workspace.Spec.Template.Attributes.GetString(constants.DevWorkspaceStorageTypeAttribute, nil)
 	return storageClass != constants.EphemeralStorageClassType
 }
+
+func addInitContainer(dwTemplateSpec *v1alpha2.DevWorkspaceTemplateSpec, initContainerFromConfig *v1alpha2.Container) {
+
+	initContainerComponent := v1alpha2.Component{
+		Name: constants.HomeInitComponentName,
+	}
+
+	initContainer := initContainerFromConfig
+
+	if initContainer == nil {
+		initContainer = inferInitContainer(dwTemplateSpec)
+	}
+
+	if initContainer == nil {
+		return
+	}
+
+	initContainerComponent.ComponentUnion = v1alpha2.ComponentUnion{
+		Container: &v1alpha2.ContainerComponent{
+			Container: *initContainer,
+		},
+	}
+
+	dwTemplateSpec.Components = append(dwTemplateSpec.Components, initContainerComponent)
+	dwTemplateSpec.Commands = append(dwTemplateSpec.Commands, v1alpha2.Command{
+		Id: constants.HomeInitEventId,
+		CommandUnion: v1alpha2.CommandUnion{
+			Apply: &v1alpha2.ApplyCommand{
+				Component: constants.HomeInitComponentName,
+			},
+		},
+	})
+	dwTemplateSpec.Events.PreStart = append(dwTemplateSpec.Events.PreStart, constants.HomeInitEventId)
+}
+
+func inferInitContainer(dwTemplateSpec *v1alpha2.DevWorkspaceTemplateSpec) *v1alpha2.Container {
+	var firstUnflattenedComponent v1alpha2.Component
+	for _, component := range dwTemplateSpec.Components {
+		if component.Container == nil {
+			continue
+		}
+
+		var err error
+		component.Attributes.Get(constants.MergedContributionsAttribute, &err)
+		if err == nil {
+			firstUnflattenedComponent = component
+			break
+		}
+	}
+
+	if firstUnflattenedComponent.Name != "" {
+		image := firstUnflattenedComponent.Container.Image
+		command := []string{"/bin/sh", "-c", "/entrypoint.sh || true"}
```

Comment:
I'm not sure how we can infer the initcontainer image and command. Maybe we can introduce a new attribute which would be added by Che, and have the attribute be configurable in the DWOC?

Some cases to think about:
- User is using a parent devfile
- User's main container image does not have the /entrypoint.sh script
- User's main container image has `["tail", "-f", "/dev/null"]` in it's entry point, which would cause the init container to never terminate

Reply:
> Maybe we can introduce a new attribute which would be added by Che, and have the attribute be configurable in the DWOC?

Can you expand upon this? 
Do you mean, having the[ initContainer image](https://github.com/devfile/devworkspace-operator/pull/1251/files#diff-1eda12737507a4bd1af2a4ea8bdaf278095ec16b208cfc2007487a507e65d36dR159-R160) and the command it runs be a Che Cluster CR field?